pynamodb/connection/base.py

```diff
@@ -1198,11 +1199,39 @@ def query(self,
         """
         operation_kwargs = {TABLE_NAME: table_name}
         name_placeholders = {}
+        expression_attribute_values = {}
+
+        tbl = self.get_meta_table(table_name)
+        if tbl is None:
+            raise TableError("No such table: {0}".format(table_name))
```

Comment:
I don't think this is valid failure case, I'd expect `get_meta_data` to raise an exception if it doesn't exist.